core/DependencyInjection/LegacyFileLoader.php

```diff
@@ -0,0 +1,208 @@
+        taoQtiTest_models_classes_QtiTestService::class,
+    ];
+
+    /**
+     * @inheritDoc
+     */
+    public function supports($resource, string $type = null)
+    {
+        return stripos($resource, '*.conf.php') !== 0;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function load($resource, string $type = null)
+    {
+        $loadClassClosure = $this->createLoadClosure();
+
+        /**
+         * @var string $path
+         * @var SplFileInfo $info
+         */
+        foreach ($this->glob($resource, false, $globResource) as $path => $info) {
+            try {
+                $class = $loadClassClosure($path);
+
+                if ($class instanceof ServiceFactoryInterface) {
+                    $this->registerInjectableClassDefinition($info);
+                }
+
+                if ($this->isLegacyClass($class)) {
+                    $this->registerClassDefinition($info, $class);
+                }
+            } finally {
+                $this->instanceof = [];
+                $this->registerAliasesForSinglyImplementedInterfaces();
+            }
+        }
+
+        $this->autoWireUnsupportedLegacyClasses();
+    }
+
+    private function registerInjectableClassDefinition(SplFileInfo $info): void
+    {
+        $alias = $this->createAlias($info);
+
+        $definition = (new Definition(ServiceFactoryInterface::class))
+            ->setAutowired(true)
+            ->setPublic(true)
+            ->setFactory(new Reference(LegacyServiceGateway::class))
+            ->setArguments([$alias]);
+
+        $this->container->setDefinition($alias, $definition);
+    }
+
+    /**
+     * @param object $class
+     */
+    private function registerClassDefinition(SplFileInfo $info, $class): void
+    {
+        $className = get_class($class);
+        $serviceName = $this->getServiceName($className);
+        $alias = $this->createAlias($info);
+
+        $definition = new Definition($serviceName);
+        $definition->setAutowired(true)
+            ->setPublic(true)
+            ->setFactory(new Reference(LegacyServiceGateway::class))
+            ->setArguments([$alias]);
+
+        $this->container->setDefinition($serviceName, $definition);
+
+        $this->container->setAlias($alias, $serviceName)
+            ->setPublic(true);
+
+        if ($className !== $serviceName) {
+            $this->container->setAlias($className, $serviceName)
+                ->setPublic(true);
+        }
+    }
+
+    private function createLoadClosure(): Closure
+    {
+        return Closure::bind(
+            function ($path) {
+                return include $path;
+            },
+            $this,
+            ProtectedPhpFileLoader::class
+        );
+    }
+
+    private function getServiceName(string $className): string
+    {
+        $interfacesWithServiceId = array_filter(
+            (new ReflectionClass($className))->getInterfaces(),
+            function (ReflectionClass $int) {
```

Comment:
It's not critical, but why it's called `int` ?
I understand, that this `int` is `interface` (I hope 😅), but during first look I thought `Hmm, why integer?`. Maybe we can rename this variable?

Reply:
Good point. FIxed!